### PR TITLE
kie-issues#1198: Copy the Compact Architecture Example in `kie-tools`

### LINKED ISSUE #1198: Copy the Compact Architecture Example into `kie-tools`
Copy the actual `Compact Architecture Example` ([process-usertasks-timer-data-index-persistence-addon-quarkus](https://github.com/apache/incubator-kie-kogito-examples/tree/main/kogito-quarkus-examples/process-usertasks-timer-data-index-persistence-addon-quarkus)) into `kie-tools` in order to be able to add showcase the `jbpm-quarkus-devui` in it.

- [ ] Copy `process-usertasks-timer-data-index-persistence-addon-quarkus` into `kie-tools` with the new bom.
- [ ] Create a `kie-tools` bom

### PR DESCRIPTION
**Pull request by pefernan**
Closes https://github.com/apache/incubator-kie-issues/issues/1198

Ensemble:
https://github.com/apache/incubator-kie-tools/pull/2391
https://github.com/apache/incubator-kie-kogito-examples/pull/1933

### CODE REVIEW

File: examples/drools-process-usertasks-quarkus-example/pom.xml [hunk 1]
```diff
@@ -32,6 +32,19 @@
...
+    <dependencies>
+      <dependency>
+        <groupId>org.jbpm</groupId>
+        <artifactId>jbpm-quarkus-devui-bom</artifactId>
+        <version>${project.version}</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
```
Comment (tiagobento) [inline examples/drools-process-usertasks-quarkus-example/pom.xml:46]: Nice!

**Comment by tiagobento** [comment]
To run `quarkus:dev` with `colima` I needed to do:
- `DOCKER_HOST="unix://${HOME}/.colima/docker.sock" pnpm quarkus:dev`.

Also, the `jbpm-quarkus-devui` doesn't work without the latest changes done to `kogito-runtimes` and `kogito-apps`, so we'll only see the updates when a new timestamped SNAPSHOT is published and we upgrade `kie-tools` to it.

File: examples/jbpm-compact-architecture-example/docker-compose/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# Kogito and Infrastructure services
+
+To allow a quick setup of all services required to run this demo, we provide a docker compose template that starts the following services:
+
+- Postgresql
+- PgAdmin
+- Kogito Data Index
```
Comment (porcelli) [inline examples/jbpm-compact-architecture-example/docker-compose/README.md]: remove this reference!

File: examples/jbpm-compact-architecture-example/docker-compose/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+```shell
+sh startServices.sh example
+```
+
+Once the services are started (depending on the profile), the following ports will be assigned on your local machine:
+
+- Postgresql: 5432
+- PgAdmin: 8055
+- Kogito Data Index: 8180
```
Comment (porcelli) [inline examples/jbpm-compact-architecture-example/docker-compose/README.md]: remove this reference!

**Review by porcelli — CHANGES_REQUESTED** [review]

**Comment by pefernan** [comment]
@porcelli done!

**Review by porcelli — APPROVED** [review]

**Comment by pefernan** [comment]
@porcelli @tiagobento don't merge yet!

**Review by tiagobento — APPROVED** [review]
Example looks good. Thanks @pefernan

**Comment by pefernan** [comment]
@porcelli @tiagobento ready!

**Pull request merged**